classes/decision/types/BackToInternalReview.inc.php

```diff
@@ -0,0 +1,224 @@
+            return;
+        }
+
+        if (!$this->canRetract($submission, $reviewRoundId)) {
+            $validator->errors()->add('restriction', __('editor.submission.decision.backToInternalReview.restriction'));
+        }
+
+        foreach ((array) $props['actions'] as $index => $action) {
+            $actionErrorKey = 'actions.' . $index;
+            switch ($action['id']) {
+                case $this->ACTION_NOTIFY_AUTHORS:
+                    $this->validateNotifyAuthorsAction($action, $actionErrorKey, $validator, $submission);
+                    break;
+            }
+        }
+    }
+
+    public function runAdditionalActions(Decision $decision, Submission $submission, User $editor, Context $context, array $actions)
+    {
+        parent::runAdditionalActions($decision, $submission, $editor, $context, $actions);
+
+        foreach ($actions as $action) {
+            switch ($action['id']) {
+                case $this->ACTION_NOTIFY_AUTHORS:
+                    $this->sendAuthorEmail(
+                        new DecisionBackToInternalReviewNotifyAuthor($context, $submission, $decision),
+                        $this->getEmailDataFromAction($action),
+                        $editor,
+                        $submission,
+                        $context
+                    );
+                    break;
+            }
+        }
+    }
+
+    public function getSteps(Submission $submission, Context $context, User $editor, ?ReviewRound $reviewRound): Steps
+    {
+        $steps = new Steps($this, $submission, $context);
+
+        $fakeDecision = $this->getFakeDecision($submission, $editor);
+        $fileAttachers = $this->getFileAttachers($submission, $context);
+
+        $authors = $steps->getStageParticipants(Role::ROLE_ID_AUTHOR);
+        if (count($authors)) {
+            $mailable = new DecisionBackToInternalReviewNotifyAuthor($context, $submission, $fakeDecision);
+            $steps->addStep(new Email(
+                $this->ACTION_NOTIFY_AUTHORS,
+                __('editor.submission.decision.notifyAuthors'),
+                __('editor.submission.decision.backToInternalReview.notifyAuthorsDescription'),
+                $authors,
+                $mailable
+                    ->sender($editor)
+                    ->recipients($authors),
+                $context->getSupportedFormLocales(),
+                $fileAttachers
+            ));
+        }
+
+        return $steps;
+    }
+
+    /**
+     * Determine if can back out to internal review stage from copy editing stage
+     */
+    public function canRetract(Submission $submission, ?int $reviewRoundId): bool
+    {
+        // If has any external review reound associated with it
+        // can not back out to internal review stage directly
+        // need to back out to external review stage first
+        if ($this->hasReviewRound($submission, WORKFLOW_STAGE_ID_EXTERNAL_REVIEW)) {
+            return false;
+        }
```

Comment:
This is needed if there is any external round exists while trying to back our form copy editing to internal review stage as for the `OMP` it's goes like `Submission --> Internal Review --> External Review --> Copy Editing` . now if there was no assigned reviewer to the only external review round when moving forward , that is to copy editing stage , it automatically get removed as that make it like `skip review and accept submission` . this check is there to prevent an accidental back out case .